docs/source/user-guide/vision/custom_checks.py

```diff
@@ -0,0 +1,209 @@
+from deepchecks.vision.context import Context, Batch
+from deepchecks.core.checks import DatasetKind
+from deepchecks.core.check_result import CheckResult
+import typing as t
+import numpy as np
+
+
+class ColorAveragesCheck(TrainTestCheck):
+    """Check if the average of each color channel is the same between the train and test dataset."""
+
+    def __init__(self, channel_names: t.Tuple[str] = None):
+        """Init the check and enable customization of the channel_names."""
+        super().__init__()
+        if channel_names is None:
+            self.channel_names = ('R', 'G', 'B')
+
+    def initialize_run(self, context: Context):
+        """Initialize the color_averages dict and pixel counter dict."""
+        self._color_averages = {
+            DatasetKind.TRAIN.value: np.zeros((3,), dtype=np.float64),
+            DatasetKind.TEST.value: np.zeros((3,), dtype=np.float64),
+        }
+        self._pixel_count = {
+            DatasetKind.TRAIN.value: 0,
+            DatasetKind.TEST.value: 0,
+        }
+
+    def update(self, context: Context, batch: Batch, dataset_kind: DatasetKind):
+        """Add the batch color counts to the color_averages dict, and update counter."""
+        images = batch.images
+        color_sums = sum(image.sum(axis=(0, 1)) for image in images)  # sum over the batch and pixels
+        self._color_averages[dataset_kind.value] += color_sums  # add to the color_averages dict
+        # count the number of pixels we have summed
+        self._pixel_count[dataset_kind.value] += sum((image.shape[0] * image.shape[1] for image in images))
+
+    def compute(self, context: Context):
+        """Compute the color averages and return them."""
+        # Divide by the number of pixels to get the average pixel value per color channel
+        for dataset_kind in DatasetKind:
+            self._color_averages[dataset_kind.value] /= self._pixel_count[dataset_kind.value]
+        # Return the color averages in a dict by channel name
+        return_value = {d_kind: dict(zip(self.channel_names, color_averages))
+                        for d_kind, color_averages in self._color_averages.items()}
+        return CheckResult(return_value)
+
+#%%
+# Hooray! we just implemented a custom check. To read more about the internal objects  Let's run it and see what happens:
+
+from deepchecks.vision.datasets.detection.coco import load_dataset
+
+train_ds = load_dataset(train=True, object_type='VisionData')
+test_ds = load_dataset(train=False, object_type='VisionData')
+
+result = ColorAveragesCheck().run(train_ds, test_ds)
+result
+
+#%%
+# Our check ran successfully, but we got the print “Nothing found”. This is because we haven’t defined to the check
+# anything to display, so the default behavior is to print “Nothing found”. In order to access the value that we have
+# defined earlier we can use the “value” property on the result.
+
+result.value
+
+#TODO: how does this LINK work?
```

Comment:
what do you mean?

Reply:
How do I link to the root of the vision API Reference?

Reply:
```
`API ref <../../api/deepchecks.vision>`_